### DISCUSSION TITLE
ERROR  Git not found during installation

### DISCUSSION THREAD

**Discussion by KwaiBird** [vladmandic/sdnext#3054] [Q&A]
Hi,

I was trying to install SD.Next, but I got "ERROR  Git not found during installation". I have already install git 2.44.0-64-bit for Windows with "Use Git from the Windows Command line and also from 3rd-party-software" checked, and python-3.10.11-amd64 (pip 24.0).
I did git clone the SD.N and run `webui.bat --debug --use-directml` with cmd.exe, but it stopped with the error message.
```
C:\[Stable Diffusion]\SD_Next_(automatic)>webui.bat --debug --use-directml
Using VENV: C:\[Stable Diffusion]\SD_Next_(automatic)\venv
15:40:49-864900 INFO     Starting SD.Next
15:40:49-869904 INFO     Logger: file="C:\[Stable Diffusion]\SD_Next_(automatic)\sdnext.log" level=DEBUG size=65
                         mode=create
15:40:49-872906 INFO     Python 3.10.11 on Windows
15:40:49-874908 ERROR    Git not found
続行するには何かキーを押してください . . .
```
[sdnext.log.zip](https://github.com/vladmandic/automatic/files/14975003/sdnext.log.zip)

How I can continue the installation?

**Additional information**
Windows 10 Home 22H2
CPU: AMD Ryzen 9 5900X
GPU: AMD Radeon RX 7900 GRE (driver: 24.3.1)
RAM: DDR4 DIMM 3200MHz 16GB 2x
Git 2.44.0-64-bit / Python 3.10.11 / pip 24.0

**Comment by brknsoul** [comment]
I'm wondering if your square brackets in your folder name might be tripping things up...

**Reply by KwaiBird** [reply]
Unfortunately, it seems the brackets are not related.

```
C:\automatic>webui.bat --debug --use-directml
Using python: "C:\Users\user\AppData\Local\Programs\Python\Python310\python.exe"
Creating VENV: C:\automatic\venv
Using VENV: C:\automatic\venv
16:53:32-410413 INFO     Starting SD.Next
16:53:32-414417 INFO     Logger: file="C:\automatic\sdnext.log" level=DEBUG size=65 mode=create
16:53:32-417419 INFO     Python 3.10.11 on Windows
16:53:32-419421 ERROR    Git not found
続行するには何かキーを押してください . . .
```

**Reply by brknsoul** [reply]
Open a new cmd prompt or Terminal window.
What does `where git` and `git --version` say?

**Reply by KwaiBird** [reply]
It looks okay.
```
C:\Users\user>where git
C:\Program Files\Git\cmd\git.exe

C:\Users\user>git --version
git version 2.44.0.windows.1
```

**Reply by brknsoul** [reply]
It sounds like you altered some options when installing git. Git installation is a next-until-done affair, where all default options are fine.

**Reply by KwaiBird** [reply]
Hmm, the issue is not fixed.
I uninstalled the Git, removed C:\Program Files\Git just in case, and reinstalled Git 2.44.0-64-bit for Windows without changing options and making sure "Use Git from ~" selected. The `where git` and `git --version` returned same strings as above.

**Reply by brknsoul** [reply]
Hrmm, I saw an issue with non-arabic characters in git's github..
You could uninstall, and try this older version;
https://github.com/git-for-windows/git/releases/download/v2.42.0.windows.2/Git-2.42.0.2-64-bit.exe

**Reply by KwaiBird** [reply]
Yes, I'm using Windows in Japanese. I should mention it first.
I have tried that version and 2.39.2.windows.1, but it does not help.
Thanks anyway. Your quick answers and confirmations help me describing the issue. I leave this open for now.

**Reply by brknsoul** [reply]
Try running
`set GIT="C:\Program Files\Git\cmd\git.exe"`
then running webui again.

**Reply by KwaiBird** [reply]
Unfortunately, this does not help.

**Comment by vladmandic** [comment]
> Git not found

this means exactly that - python cannot find `git`.
try executing this manually - its basically same to what sdnext does:

```shell
C:\>python
Python 3.11.9 (tags/v3.11.9:de54cf5, Apr  2 2024, 10:12:12) [MSC v.1938 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import shutil
>>> shutil.which('git')
'c:\\program files\\git\\cmd\\git.EXE'
>>>
```

and if you think git is really there and want to ignore this check, you can simply append `--ignore` cmd flag to webui.bat.

**Reply by KwaiBird** [reply]
Thanks. However, the same error occurred again, and I got some new errors.

I Installed Python 3.11.9, uninstalled Git and installed `Git-2.44.0-64-bit.exe` again, and rebooted my system.
```
 c:\>where git && git --version
C:\Program Files\Git\cmd\git.exe
git version 2.44.0.windows.1
```
```
 c:\>git clone https://github.com/vladmandic/automatic
Cloning into 'automatic'...
remote: Enumerating objects: 43528, done.
remote: Counting objects: 100% (117/117), done.
remote: Compressing objects: 100% (65/65), done.
remote: Total 43528 (delta 57), reused 107 (delta 51), pack-reused 43411
Receiving objects: 100% (43528/43528), 53.72 MiB | 24.12 MiB/s, done.
Resolving deltas: 100% (31419/31419), done.

c:\>cd automatic

c:\automatic>webui.bat --debug --use-directml
Using python: "C:\Users\user\AppData\Local\Programs\Python\Python310\python.exe"
Creating VENV: c:\automatic\venv
Using VENV: c:\automatic\venv
14:34:00-285840 INFO     Starting SD.Next
14:34:00-289844 INFO     Logger: file="c:\automatic\sdnext.log" level=DEBUG size=65 mode=create
14:34:00-291846 INFO     Python 3.10.11 on Windows
14:34:00-293847 ERROR    Git not found
続行するには何かキーを押してください . . .
```

Then, I did this commands manually.

```
c:\>python
Python 3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import shutil
>>> shutil.which('git')
'C:\\Program Files\\Git\\cmd\\git.EXE'
>>>
```
but I still got the same error.

```
C:\automatic>webui.bat --debug --use-directml
Using VENV: C:\automatic\venv
14:40:36-221358 INFO     Starting SD.Next
14:40:36-226363 INFO     Logger: file="C:\automatic\sdnext.log" level=DEBUG size=65 mode=create
14:40:36-229366 INFO     Python 3.10.11 on Windows
14:40:36-230367 ERROR    Git not found
続行するには何かキーを押してください . . .
```
So, I did `webui.bat --debug --use-directml  --ignore`, but it halted with some errors.
Since some words in the log are corrupted and readable, I changed my Windows language settings from Japanese to English.
Then run `webui.bat --debug --use-directml  --ignore` again. Same things happened again.
[snnext_logs.zip](https://github.com/vladmandic/automatic/files/15020580/snnext_logs.zip)

P.S. It seems that my system does not use Python 3.11.9 as default but uses Python 3.10.11 instead.
I have 3 Python versions in the system, I hope it does not make troubles.
```
C:\Users\user>py --list-paths
Installed Pythons found by py Launcher for Windows
 -3.12-64       C:\Users\user\miniconda3\python.exe *
 -3.11-64       C:\Users\user\AppData\Local\Programs\Python\Python311\python.exe
 -3.10-64       C:\Users\user\AppData\Local\Programs\Python\Python310\python.exe
```

```
C:\Users\user>python --version
Python 3.10.11

C:\Users\user>pip --version
pip 24.0 from C:\Users\user\AppData\Local\Programs\Python\Python310\lib\site-packages\pip (python 3.10)
```

**Comment by TangentLe** [comment]
https://stackoverflow.com/questions/26620312/git-installing-git-in-path-with-github-client-for-windows
